services/userservice/userservice.go

```diff
@@ -1,11 +1,14 @@
 package userservice
 
 import (
+	"errors"
 	"github.com/atmiguel/cerealnotes/databaseutil"
 	"golang.org/x/crypto/bcrypt"
 	"time"
 )
 
+var EmailAddressAlreadyInUseError = errors.New("")
```

Comment:
Why not put an error message here instead of empty string?

Reply:
Also in my PR i added https://github.com/pkg/errors. With this Library creating errors is easier. But I think you'll need to create a function that returns a new error. Otherwise the stacktrace for this error will always be line 10 in userservice

Reply:
I suppose I was just thinking that the error message should never be displayed, because this error should always be checked for, but that's not a great reason. I'll add a message.

I'm not clear on what pkg/errors does. Looks like it makes it easier to give context to custom errors? We don't need context here.

Also, I'm not sure what you mean by creating a function that returns a new error?

Reply:
I suppose I could add which email address it is to the context. Though again, the expectation is that the error is caught immediately

Reply:
I did some testing and we can ignore the function part, since normal go errors don't have any stacktrace-like information.